### PR TITLE
Ensure that dynamic records map to unique Azure TM profiles

### PR DESCRIPTION
**Pull request by viranch**
... and throw exception with a collision is detected.

### CODE REVIEW

File: octodns/provider/azuredns.py [hunk 1]
```diff
@@ -259,8 +259,21 @@ def _parse_azure_type(string):
     return string.split('/')[-1]
 
 
-def _traffic_manager_suffix(record):
-    return record.fqdn[:-1].replace('.', '-')
+def _root_traffic_manager_name(record):
+    # ATM names can only have letters, numbers and hyphens
+    # replace dots with double hyphens to ensure unique mapping,
+    # hoping that real life FQDNs won't have double hyphens
+    return record.fqdn[:-1].replace('.', '--')
```
Comment (ross) [inline octodns/provider/azuredns.py:266]: Naming character restrictions like this are so annoying. Computers don't care what characters are in there... :grin:

File: octodns/provider/azuredns.py [hunk 2]
```diff
@@ -613,8 +626,13 @@ def _data_for_CNAME(self, azrecord):
...
             :type  return: dict
         '''
-        if azrecord.cname_record is None and azrecord.target_resource.id:
-            return self._data_for_dynamic(azrecord)
+        if azrecord.cname_record is None:
+            if azrecord.target_resource.id:
+                return self._data_for_dynamic(azrecord)
+            else:
+                # dynamic record alias is broken, return dummy value and apply
+                # will likely overwrite/fix it
```
Comment (ross) [inline octodns/provider/azuredns.py:634]: Probably worth a warning log here with a bit more info about what's happening, like the comment.

**Review by ross — COMMENTED** [review]
Looks good overall. Comment/suggestion inline.

File: octodns/provider/azuredns.py (hunk 2 above)
Comment (viranch) [reply octodns/provider/azuredns.py:634]: pushed new commit for this

File: octodns/provider/azuredns.py (hunk 1 above)
Comment (viranch) [reply octodns/provider/azuredns.py:266]: There go my plans of using emojis for traffic manager names 👎

**Review by ross — APPROVED** [review]

**Pull request merged**